backends-velox/src/test/scala/org/apache/gluten/execution/VeloxCacheSuite.scala

```diff
@@ -0,0 +1,93 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.execution
+
+import org.apache.gluten.GlutenConfig
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class VeloxCacheSuite extends AnyFunSuite with AdaptiveSparkPlanHelper {
+
+  /**
+   * Velox cache is controlled by static configuration, this method is used to create a new
+   * SparkSession.
+   */
+  def withSparkSession(f: SparkSession => Unit, cacheEnabled: Boolean): Unit = {
+    val spark = SparkSession
+      .builder()
+      .master("local[1]")
+      .appName(s"test-veloxCache_$cacheEnabled")
+      .config(GlutenConfig.COLUMNAR_VELOX_CACHE_ENABLED.key, cacheEnabled.toString)
+      .config(GlutenConfig.COLUMNAR_VELOX_FILE_HANDLE_CACHE_ENABLED.key, cacheEnabled.toString)
+      .config(GlutenConfig.LOAD_QUANTUM.key, "8MB")
+      .config("spark.plugins", "org.apache.gluten.GlutenPlugin")
+      .config("spark.default.parallelism", "1")
+      .config("spark.memory.offHeap.enabled", "true")
+      .config("spark.memory.offHeap.size", "1024MB")
+      .config("spark.ui.enabled", "false")
+      .config("spark.gluten.ui.enabled", "false")
+      .getOrCreate()
+    try f(spark)
+    finally {
+      spark.stop()
+    }
+  }
+
+  test("Velox cache metrics") {
+    def test(cacheEnabled: Boolean)(spark: SparkSession): Unit = {
+      val table = s"metrics_t1_$cacheEnabled"
+      try {
+        spark.range(10).write.saveAsTable(table)
+
+        // first scan, read from storage.
+        val df = spark.sql(s"SELECT * FROM $table")
+        val scans = collect(df.queryExecution.executedPlan) {
+          case scan: FileSourceScanExecTransformer => scan
+        }
+        df.collect()
+        assert(scans.length === 1)
+        val metrics = scans.head.metrics
+        assert(metrics("storageReadBytes").value > 0)
+        assert(metrics("ramReadBytes").value == 0)
+
+        // second scan, read from cache if cache enabled.
+        val df2 = spark.sql(s"SELECT * FROM $table")
+        val scans2 = collect(df2.queryExecution.executedPlan) {
+          case scan: FileSourceScanExecTransformer => scan
+        }
+        df2.collect()
+        assert(scans2.length === 1)
+        val metrics2 = scans2.head.metrics
+        if (cacheEnabled) {
+          assert(metrics2("storageReadBytes").value == 0)
+          assert(metrics2("ramReadBytes").value > 0)
+        } else {
+          assert(metrics2("storageReadBytes").value > 0)
+          assert(metrics2("ramReadBytes").value == 0)
+        }
+      } finally {
+        spark.sql(s"DROP TABLE IF EXISTS $table")
+      }
+    }
+
+    Seq("true", "false").foreach(
```

Comment:
this UT will fail due to the current lack of a shutdown, which will be fixed in another PR.